tests/integration/deploy/test_deploy_command.py

```diff
@@ -423,6 +423,30 @@ def test_deploy_without_s3_bucket(self, template_file):
             deploy_process_execute.stderr,
         )
 
+    @parameterized.expand(["aws-serverless-function.yaml", "cdk_v1_synthesized_template_zip_functions.json"])
+    def test_deploy_with_no_resolve_s3_option(self, template_file):
+        template_path = self.test_data_path.joinpath(template_file)
+
+        stack_name = self._method_to_stack_name(self.id())
+
+        deploy_command_list = self.get_deploy_command_list(
+            template_file=template_path,
+            stack_name=stack_name,
+            capabilities="CAPABILITY_IAM",
+            s3_prefix=self.s3_prefix,
+            no_resolve_s3=True,
+            force_upload=True,
+            notification_arns=self.sns_arn,
+            parameter_overrides="Parameter=Clarity",
+            kms_key_id=self.kms_key,
+            no_execute_changeset=False,
+            tags="integ=true clarity=yes foo_bar=baz",
+            confirm_changeset=False,
+        )
+
+        deploy_process_execute = self.run_command(deploy_command_list)
+        self.assertEqual(deploy_process_execute.process.returncode, 0)
```

Comment:
building from my top level comment, we can check for the error message about missing bucket instead, like this test: https://github.com/aws/aws-sam-cli/blob/e0bc44db64c9e81ea25521ed5cbd8fc8d2245c9c/tests/integration/deploy/test_deploy_command.py#L393-L424